packages/paymasters/contracts/PermitERC20UniswapV3Paymaster.sol

```diff
@@ -269,11 +275,7 @@ contract PermitERC20UniswapV3Paymaster is BasePaymaster, ERC2771Recipient {
             }
             UniswapV3Helper.unwrapWeth(uniswap, amountSwapped);
         }
-        if (balance + amountSwapped < depositAmount) {
-            emit FundingNeeded();
-            depositAmount = balance + amountSwapped;
-        }
-        relayHub.depositFor{value : depositAmount}(address(this));
+        relayHub.depositFor{value : balance + amountSwapped}(address(this));
```

Comment:
Or, do the `{value:  address(this).balance}`. This is boolet-proof while `balance + amountSwapped` really depends on `amountSwapped` always being correct, which is an assumption that 3rd party solidity code is precise to a gwei.

Reply:
It was simply cheaper that way